lang/java/avro/src/main/java/org/apache/avro/Schema.java

```diff
@@ -1546,26 +1461,13 @@ public Schema parse(File file) throws IOException {
       return parse(FACTORY.createParser(file), false);
     }
 
-    public List<Schema> parse(Iterable<File> sources) throws IOException {
```

Comment:
I'm very concerned about breaking public API's

Reply:
This bit of public API was added after the last release (#2375).

Other than that, your concern is valid: we should not break a public API before at least having it deprecated, with a documented alternative, for sufficient time to migrate (probably at least a year).